coax/utils/_misc.py

```diff
@@ -472,11 +472,13 @@ def generate_gif(env, filepath, policy=None, resize_to=None, duration=50, max_ep
     if isinstance(env, TrainMonitor):
         env = env.env  # unwrap to strip off TrainMonitor
 
-    assert env.render_mode == 'rgb_array', "env.render_mode must be 'rgb_array'"
+    s, info = env.reset()
+
+    # check if render_mode is set to 'rbg_array'
+    assert isinstance(env.render(), onp.ndarray), "env.render_mode must be set to 'rgb_array'"
```

Comment:
Please could you change the check to something like:
```
if env.render_mode != 'rgb_array' and not isinstance(env.render(), onp.ndarray):
  raise RuntimeError("Cannot generate GIF if env.render_mode != 'rgb_array'.")
```
This avoids unnecessary rendering if the render_mode is set properly.

Thanks!

Reply:
Just added the commit expanding the conditional.
Thanks! :)